File: C++ Programs/Algorithms/SearchingAlgo/LinearSearch.cpp
#include <iostream>
#include <vector>
using namespace std;

int linear_search(vector<int> list, int x)
{
    for (int i = 0; i < list.size(); i++)
    {
        if (list[i] == x)
            return i;
        else
            continue;
    }
    return -1;
}

int main()
{
    int size, ele, data = 0, pos;
    vector<int> list;
    cout << "Enter size of the list : ";
    cin >> size;
    cout << "Enter elements for the list : ";
    for (int i = 0; i < size; i++)
    {
        cin >> data;
        list.push_back(data);
    }
    cout << "Enter element to search in the list : ";
    cin >> ele;
    pos = linear_search(list, ele);
    if (pos >= 0)
        cout << "Element found at position " << pos+1;
    else
        cout << "Element does not exist in the list.";
    
    return 0;
}

File: Java_Programs/Algorithms/SearchingAlgo/LinearSearch.java
package Java_Programs.Algorithms.SearchingAlgo;

import java.util.Scanner;

public class LinearSearch {
    static int linearSearch(int arr[], int x) {
        for (int i = 0; i < arr.length; i++)
        {
            if (arr[i] == x)
                return i;
            else
                continue;
        }
        return -1;
    }

    public static void main(String[] args) {
        Scanner x = new Scanner(System.in);
        System.out.print("Enter size of array : ");
        int size = x.nextInt();
        int arr[] = new int[size];
        System.out.print("\nEnter elements : ");
        for (int i = 0; i < size; i++) {
            arr[i] = x.nextInt();
        }

        System.out.print("Enter element to search for : ");
        int ele = x.nextInt();
        int pos = linearSearch(arr, ele);
        if (pos >= 0)
            System.out.print("\nElement found in position " + (pos+1));
        else
            System.out.print("\nElement doesn't exist in the array.");
        x.close();
    }
}
